GoogleDataTransport/GDTCORLibrary/GDTCORMetricsController.m

```diff
@@ -114,22 +104,25 @@ - (instancetype)initWithStorage:(id<GDTCORStoragePromiseProtocol>)storage {
                                               eventMetricsCounter:nil];
   };
 
-  return [_storage fetchAndUpdateLibraryDataForKey:kMetricsLibraryDataKey
-                                             klass:[GDTCORMetricsMetadata class]
-                                    readWriteBlock:readWriteblock]
-      .then(^id _Nullable(GDTCORLibraryData _Nullable value) {
-        // TODO(ncooke3): Create metrics object using snapshottedMetadata.
-        return nil;
+  return FBLPromise
+      .all(@[
+        [_storage fetchAndUpdateClientMetricsWithReadWriteBlock:readWriteblock],
+        [_storage fetchStorageMetadata]
+      ])
+      .then(^GDTCORMetrics *(NSArray *metricsMetadataAndStorageMetadata) {
+        return
+            [GDTCORMetrics metricsWithMetricsMetadata:metricsMetadataAndStorageMetadata.firstObject
+                                      storageMetadata:metricsMetadataAndStorageMetadata.lastObject];
       });
 }
 
 - (nonnull FBLPromise<NSNull *> *)offerMetrics:(nonnull GDTCORMetrics *)metrics {
-  GDTCORStorageLibraryDataReadWriteBlock readWriteblock = ^GDTCORMetricsMetadata *(
+  __auto_type readWriteblock = ^GDTCORMetricsMetadata *(
```

Comment:
maybe rename it to fetchAndUpdateCompleteHandler to make it more consistent with function name and easier to understand

Reply:
Done in 1c37aa6